### PR TITLE
Add Job submission limit

### PR DESCRIPTION
**Pull request by Gnodesign**
### Changes proposed in this Pull Request

* Just like in the Resume Manager, users can set a job submission limit via the settings.
* Adds a "post job" in the job dashboard if the **submission limit** is bigger than the submitted listings.
* Redirects user to the **Job Dashboard** from the **Job Submission** page if the submission limit has been exceeded

### Testing instructions

* Go to "Job Listings -> Settings -> Job Submission" and set a value for the "Listing limit"
* Go to **Submit Job Form Page** and you will be redirect to the dashboard if the limit has been exceeded

The behavior is exactly the same as in the **Resume Manager** plugin.

### CODE REVIEW

**Comment by Gnodesign** [comment]
Lots of users have been asking for this feature, it would be great to have this added in the next release. Could someone review this?
@jom @donnapep @renatho @alexsanford

**Comment by Gnodesign** [comment]
Since all these PRs are getting merged maybe we can merge this as well. :) 
@renatho @fjorgemota @onubrooks

File: includes/class-wp-job-manager-shortcodes.php [hunk 1]
```diff
@@ -90,6 +91,56 @@ private function is_job_dashboard_page() {
...
+			 */
+			$redirect_url = apply_filters(
+				'job_manager_redirect_url_exceeded_listing_limit',
+				$redirect_url,
+				$submission_limit,
+				$job_count
+			);
+
+			if ( $redirect_url ) {
+				wp_safe_redirect( esc_url( $redirect_url ) );
```
Comment (onubrooks) [inline includes/class-wp-job-manager-shortcodes.php:137]: Do you think we can add a notice explaining the reason for the redirect?

**Review by onubrooks — APPROVED** [review]
Works for me, added a comment below though.

**Pull request merged**